test/lint/lint-cppcheck.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+export LC_ALL=C
+
+ENABLED_CHECKS=(
+    "Class '.*' has a constructor with 1 argument that is not explicit."
+    "Struct '.*' has a constructor with 1 argument that is not explicit."
+)
+
+IGNORED_WARNINGS=(
+    "src/arith_uint256.h:.* Class 'arith_uint256' has a constructor with 1 argument that is not explicit."
+    "src/arith_uint256.h:.* Class 'base_uint < 256 >' has a constructor with 1 argument that is not explicit."
+    "src/arith_uint256.h:.* Class 'base_uint' has a constructor with 1 argument that is not explicit."
+    "src/net.h:.* Class 'CNetMessage' has a constructor with 1 argument that is not explicit."
+    "src/prevector.h:.* Class 'const_iterator' has a constructor with 1 argument that is not explicit."
+    "src/prevector.h:.* Class 'const_reverse_iterator' has a constructor with 1 argument that is not explicit."
+    "src/prevector.h:.* Class 'iterator' has a constructor with 1 argument that is not explicit."
+    "src/prevector.h:.* Class 'reverse_iterator' has a constructor with 1 argument that is not explicit."
+    "src/primitives/transaction.h:.* Class 'CTransaction' has a constructor with 1 argument that is not explicit."
+    "src/protocol.h:.* Class 'CMessageHeader' has a constructor with 1 argument that is not explicit."
+    "src/rpc/server.h:.* Struct 'UniValueType' has a constructor with 1 argument that is not explicit."
```

Comment:
Why are some of our own classes suppressed as warnings and others fixed up? Seems arbitrary to me.

Reply:
I excluded the constructors that appeared to be intentionally non-`explicit` or where switching to `explicit` seemed non-trivial.

The remaining cases are:

```
CTransaction(CMutableTransaction &&tx)
CTransaction(const CMutableTransaction &tx)
UniValueType(UniValue::VType _type)
arith_uint256(const base_uint<256>& b)
arith_uint256(uint64_t b)
base_uint(uint64_t b)
const_iterator(const T* ptr_)
const_iterator(iterator x)
const_reverse_iterator(const T* ptr_)
const_reverse_iterator(reverse_iterator x)
iterator(T* ptr_)
reverse_iterator(T* ptr_)
secure_allocator(const secure_allocator<U>& a)
zero_after_free_allocator(const zero_after_free_allocator<U>& a)
```

And also two cases that appear to be incorrectly flagged as single-parameter:

```
CMessageHeader(const MessageStartChars& pchMessageStartIn, const char* pszCommand, unsigned int nMessageSizeIn)
CNetMessage(const CMessageHeader::MessageStartChars& pchMessageStartIn, int nTypeIn, int nVersionIn)
```

Should any of these be be turned `explicit` as part of this PR? :-)

Reply:
Makes sense. Though, I'd prefer if this was a test only change (adding all the suppressions) and then go out and fixup violations in the code in another pull request(s), since both tasks are slightly different when it comes to review process.